tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -2644,80 +2644,15 @@ uint32_t EmitTypeHandler::emitType(const SpirvType *type) {
   // NodePayloadArray types
   else if (const auto *npaType = dyn_cast<NodePayloadArrayType>(type)) {
     const uint32_t elemTypeId = emitType(npaType->getElementType());
+
+    // Output the decorations for the type first. This will create other values
+    // that are on the decorations, and they must appear before the type.
+    emitDecorationsForNodePayloadArrayTypes(npaType, id);
```

Comment:
I suppose this was moved for the same reason? But no test change seem related to the node payload?

Reply:
Yes it is. Some tests failed validation with the latest spirv-tools without this. This change will be tested once we update spirv-tools. I want it to be a separate change in case there is a regression. It will be easier to identify what caused the change.